core/store/src/db.rs

```diff
@@ -475,7 +474,11 @@ impl RocksDBOptions {
     }
 
     /// Opens the database in read/write mode.
-    pub fn read_write<P: AsRef<std::path::Path>>(self, path: P) -> Result<RocksDB, DBError> {
+    pub fn read_write<P: AsRef<std::path::Path>>(
+        self,
+        path: P,
+        extra_db_memory: usize,
+    ) -> Result<RocksDB, DBError> {
```

Comment:
This is a builder object, please add  `fn extra_db_memory(self, amount: bytesize::ByteSize) -> Self` rather than a separate parameter.